Project 2/k-NNAlgorithms/src/Data.java

```diff
@@ -1,2 +1,296 @@
+import java.io.File;
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.Scanner;
+
 public class Data {
+    ArrayList<ArrayList<ArrayList<Double>>> splitData= new ArrayList<ArrayList<ArrayList<Double>>>();
+    ArrayList<ArrayList<Double>> fullSet = new ArrayList<ArrayList<Double>>();
+    String [][] inputFileAsArray;
+    String dataName;
+
+    Data(File inputFileName, String dataName) throws Exception{
+        this.dataName=dataName;
+        this.inputFileAsArray=fileTo2dString(inputFileName);
+        preprocessData();
+        convertInputFileToArrayList();
+        bucketizeData();
+    }
+
+    ///////////////////////////////////////////////////////////////
+    //function for input data to be changed into a 2d string array
+    private  String [][] fileTo2dString(File inputFile) throws Exception{
+
+        //determines the dimensions of the file
+        int count=0;
+        Scanner sc = new Scanner(inputFile);
+        while (sc.hasNextLine()){
+            count++;
+            sc.nextLine();
+        }
+        sc.close();
+        sc = new Scanner(inputFile);
+        String [] line= sc.nextLine().split(",");
+
+        //once dimensions are found and 2d array data made, first line is added
+        String [][] data= new String [count][line.length];
+        //count is reset to increment the row while filling the array
+        count=0;
+        data[0]=line;
+        //rest of file is filled into the 2d array
+        while (sc.hasNextLine()){
+            count++;
+            data[count]= sc.nextLine().split(",");
+        }
+
+
+
+
+        //randomize the order of the input array
+        //by swapping rows n times
+        for (int i = 0; i < data.length ; i++) {
+            //generate random number within range of data
+            int randomNum= (int)Math.floor((Math.random()*data.length));
+
+            for (int j = 0; j <data[0].length ; j++) {
+                //swaps data in random row and current row
+                String temp= data[i][j];
+                data[i][j]=data[randomNum][j];
+                data[randomNum][j]=temp;
+            }
+        }
+        sc.close();
+        return data;
+    }
+
+
+    private void preprocessData(){
+
+        //switch for data input
+        switch(dataName){
```

Comment:
If there's a way to use an interface or something here with an abstract preprocessData() for each type of data, we should do it. If not, this works, it's just big.